Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -446,15 +447,15 @@ void LeBailFit::processInputBackground() {
 
 //----------------------------------------------------------------------------------------------
 /** Calcualte LeBail diffraction pattern:
- *  Output spectra:
- *  0: data;  1: calculated pattern; 3: difference
- *  4: input pattern w/o background
- *  5~5+(N-1): optional individual peak
- */
+*  Output spectra:
+*  0: data;  1: calculated pattern; 3: difference
+*  4: input pattern w/o background
+*  5~5+(N-1): optional individual peak
+*/
 void LeBailFit::execPatternCalculation() {
   // Generate domain and values vectors
-  const MantidVec &vecX = m_dataWS->readX(m_wsIndex);
-  MantidVec &vecY = m_outputWS->dataY(CALDATAINDEX);
+  const auto &vecX = m_dataWS->x(m_wsIndex).rawData();
+  std::vector<double> vecY(m_outputWS->y(CALDATAINDEX).size(), 0);
```

Comment:
Why is this being zero-initialized now, instead of copied? Is it just overwritten below?